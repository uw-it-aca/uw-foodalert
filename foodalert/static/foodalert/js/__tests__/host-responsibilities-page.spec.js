"use strict";
import Vue from 'vue';
import BootstrapVue from 'bootstrap-vue'
Vue.use(BootstrapVue);

import {mount, shallowMount} from '@vue/test-utils';
import ResponsibilitiesPage from '../pages/host/responsibilities-page.vue';

const $router = {
    push: jest.fn(),
}

describe('Food Categories Page', function () {
        test('renders correctly', function () {
            var wrapper = mount(ResponsibilitiesPage, {
            });
            expect(wrapper.element).toMatchSnapshot();
        });
	
	test('testing options - no option selected', function () {
	    var wrapper = mount(ResponsibilitiesPage, {
	        mocks: { $router }
	    });
	    const button = wrapper.findAll('button').at(0)
	    button.trigger('click')
	});

	test('testing options - first option selected', function () {
	    var wrapper = mount(ResponsibilitiesPage, {
	        mocks: { $router }
	    });
            const checkboxes = wrapper.findAll('input[type="checkbox"]') 
	    checkboxes.at(0).setChecked()
	    const button = wrapper.findAll('button').at(0)
	    button.trigger('click')
	});

	test('testing options - second option selected', function () {
	    var wrapper = mount(ResponsibilitiesPage, {
	        mocks: { $router }
	    });
            const checkboxes = wrapper.findAll('input[type="checkbox"]') 
	    checkboxes.at(1).setChecked()
	    const button = wrapper.findAll('button').at(0)
	    button.trigger('click')
	});

	test('testing options - both options selected', function () {
	    var wrapper = mount(ResponsibilitiesPage, {
	        mocks: { $router }
	    });
            const checkboxes = wrapper.findAll('input[type="checkbox"]') 
	    checkboxes.at(0).setChecked()
	    checkboxes.at(1).setChecked()
	    const button = wrapper.findAll('button').at(0)
	    button.trigger('click')
	    const form = wrapper.find('form')
	    form.trigger('submit')
	});

	test('back button', function () {
	    var wrapper = mount(ResponsibilitiesPage, {
	        mocks: { $router }
	    });
	    const button = wrapper.findAll('button').at(1)
	    button.trigger('click')
	});
})
