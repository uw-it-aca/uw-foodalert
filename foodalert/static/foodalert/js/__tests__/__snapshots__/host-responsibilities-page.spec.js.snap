// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Food Categories Page renders correctly 1`] = `
<div
  class="page"
>
  <div
    id="standard-notification"
  >
    <div
      class="collapse"
      id="notif-container"
      style="display: none;"
    >
      <div
        class="text-center py-3 container-fluid"
      >
        <div
          class="container"
        >
          <div
            class="row justify-content-center"
          >
            <div
              class="col-md-8"
            >
              <div
                class="h5 text-white pt-2"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="page-content pb-2"
  >
     
    <div
      class="container"
    >
      <div
        class="row justify-content-center"
      >
        <div
          class="page-content-padding col-md-12 col-lg-8"
        >
          <h1
            class="pt-md-5 pt-3 pb-2"
            id="standard-heading"
          >
            
        Responsibilities
    
          </h1>
           
          <div
            id="standard-body"
          >
            <p>
              
          To ensure the safety of the food being shared, we ask that you
          to agree to all of the following requirements.
        
            </p>
             
            <form
              class=""
            >
              <div
                class="mt-2 custom-control custom-checkbox"
              >
                <input
                  autocomplete="off"
                  class="custom-control-input"
                  type="checkbox"
                  value="cond1"
                />
                <label
                  class="custom-control-label"
                >
                  <span>
                    
              My office is responsible for the safety of this food.
              
                    <a
                      aria-controls="cond1-info"
                      aria-expanded="false"
                      aria-label="Learn more about safety responsibilities"
                      class=""
                      herf="#"
                      href="#"
                      role="button"
                      target="_self"
                    >
                      
                Learn more
              
                    </a>
                  </span>
                </label>
              </div>
               
              <div
                class="custom-collapse collapse-text-box collapse"
                id="cond1-info"
                style="display: none;"
              >
                <div
                  class="custom-text-box"
                >
                  
            You are responsible for food safety under the approval of
            the Temporary Food Permit. This same level of responsibility
            applies when using the Hungry Husky App.
          
                </div>
              </div>
               
              <div
                class="mt-2 custom-control custom-checkbox"
              >
                <input
                  autocomplete="off"
                  class="custom-control-input"
                  type="checkbox"
                  value="cond2"
                />
                <label
                  class="custom-control-label"
                >
                  <span>
                    
              Potentially hazardous food must be kept at proper
              temperatures and have been sitting at room temperature
              for no more than 
                    <strong>
                      four hours
                    </strong>
                    .
              
                    <a
                      aria-controls="cond2-info"
                      aria-expanded="false"
                      aria-label="Learn more about potentially hazardous food"
                      class=""
                      herf="#"
                      href="#"
                      role="button"
                      target="_self"
                    >
                      
                Learn more
              
                    </a>
                  </span>
                </label>
              </div>
               
              <div
                class="custom-collapse collapse-text-box collapse"
                id="cond2-info"
                style="display: none;"
              >
                <div
                  class="custom-text-box"
                >
                  
              Information is missing
          
                </div>
              </div>
               
              <div
                class="mt-5"
              >
                <div
                  class="row justify-content-between"
                >
                  <div
                    class="col-md-4 col-lg-3 order-md-2"
                  >
                    <button
                      class="btn mb-3 btn-primary btn-lg btn-block disabled"
                      disabled="disabled"
                      type="submit"
                    >
                      
                        Agree
                    
                    </button>
                  </div>
                   
                  <div
                    class="col-md-4 col-lg-3 order-md-1"
                  >
                    <button
                      class="btn hh-back-button btn-outline-secondary btn-lg btn-block"
                      type="submit"
                    >
                      
                        Back
                    
                    </button>
                  </div>
                </div>
              </div>
            </form>
          </div>
           
        </div>
      </div>
    </div>
  </div>
   
  <footer
    class="text-center"
    id="relative-footer"
  >
    <a
      href="mailto:help@uw.edu?subject=Hungry Husky support"
    >
      
        Contact support
      
    </a>
     
    <p>
      Â© 2019 University of Washington
    </p>
  </footer>
</div>
`;
